gbfs.md

```diff
@@ -101,9 +101,17 @@ Producers SHOULD provide a technical contact who can respond to feed outages in
 
 ### Seasonal Shutdowns, Disruptions of Service
 
-Feeds SHOULD continue to be published during seasonal or temporary shutdowns.  Feed URLs SHOULD NOT return a 404.  An empty bikes array SHOULD be returned by `free_bike_status.json`. Stations in `station_status.json` SHOULD be set to `is_renting:false`, `is_returning:false` and `is_installed:false` where applicable. Seasonal shutdown dates SHOULD be reflected in `system_calendar.json`.
+Feeds SHOULD continue to be published during seasonal or temporary shutdowns.  Feed URLs SHOULD NOT return a 404.  An empty `bikes` array SHOULD be returned by `free_bike_status.json`. Stations in `station_status.json` SHOULD be set to `is_renting:false`, `is_returning:false` and `is_installed:false` where applicable. Seasonal shutdown dates SHOULD be reflected using `opening_hours` in `system_information.json`.
 
 Announcements for disruptions of service, including disabled stations or temporary closures of stations or systems SHOULD be made in `system_alerts.json`.
+###  Hours and Dates of Operation
+Beginning with v3.0, hours and dates of operation are described using the Open Street Map [opening_hours](https://wiki.openstreetmap.org/wiki/Key:opening_hours) format. The OSM opening_hours syntax is quite complex, therefore it is RECOMMENDED that publishers validate their opening_hours data to ensure its accuaracy.
+* [OSM opening_hours examples](https://wiki.openstreetmap.org/wiki/Key:opening_hours)
+* [OSM opening_hours syntax guide](https://wiki.openstreetmap.org/wiki/Key:opening_hours/specification)
+* [OSM opening_hours validation tool](https://openingh.openstreetmap.de/evaluation_tool/)
+* [OSM opening_hours project and code libraries](https://github.com/opening-hours)
+
+Hours and dates of operation SHOULD be published even in cases where services are continuously available 24/7. During periods when a system or station is outside of opening hours, stations SHOULD be set to `is_renting = false`. During these periods, `station_status.json.num_bikes_available` and `station_status.json.num_docks_available` SHOULD reflect the number of vehicles and docks that would be available if the system or station were open.
```

Comment:
Hi, could this clarify how free_bike_status.json should behave wrt. opening hours since that feed doesn't have is_renting field? I.e. if a system is outside of opening hours, would there be no free bikes? Also, in this paragraph stations SHOULD be set to `is_returning=false` too outside of opening hours.

We're trying to figure out whether we MUST check system_information.opening_hours when ingesting real-time feeds before surfacing bikes (i.e. this is a breaking proposal) or we can continue gating by is_renting=false, is_returning=false and no bikes in free_bike_status.

Reply:
if a system is outside of opening hours, there would not be any vehicles available for rent, including free floating bikes. This can be added to the system_information.opening_hours definition.

> Also, in this paragraph stations SHOULD be set to is_returning=false too outside of opening hours.

Actually this is not the case - systems typically allow you to end a rental and/or return vehicles outside of opening hours.  

> We're trying to figure out whether we MUST check system_information.opening_hours when ingesting real-time feeds before surfacing bikes (i.e. this is a breaking proposal) or we can continue gating by is_renting=false, is_returning=false and no bikes in free_bike_status. 

This one is harder to answer - It seems like for now, you would have to check `system_information.opening_hours` in the case of free bikes. In  `station_status` ,  the `num_bikes_available` and `num_docks_available` fields give the numbers that would be available if the station were set to allow rentals. There's a long thread on how this came to be in #94 . The reason was so that you could plan a trip or reserve a vehicle at 5:45 when the system opened at 6:00.  There's no equivalent to `num_bikes_available` in `free_bike_status` but there's also no where that it says there should be no bikes in `free_bike_status` when rentals aren't allowed, only when there are no bikes on the street.  If there are bikes on the street, they could still appear in `free_bike_status` when `opening_hours` evaluates to `closed`, that's no different from the current spec, it's just that the format and location of the hours of operation are different.

Reply:
I see. There's also `is_disabled` field in free_bike_status which says "This field is used to indicate vehicles that are in the field but not available for rental." It sounds like this field should then be set to false outside of opening hours?

Reply:
I don't know if I agree with that. With station based ( docked) vehicles that's not really how it works and I think it should be the same for free floating.  The numbers represented by `ststion_status.num_bikes_available` and `num_docks_available` are the number of functional bikes/docks that would be available if `is_renting` = `true`. Suppose there's a free floating vehicle with a dead battery and you set 'is_disabled`= `true'  when the system is closed. In the above scenario where you're planning a trip at 5:45 for the system that opens at 6:00, it would appear that this vehicle will be available for rent at 6 but it won't be.

Reply:
There should be a separate PR to improve documentation / meaning of the is_disabled wrt opening hours.